scripts/nyc/transform-coverage.ts

```diff
@@ -54,8 +62,24 @@ export function transformCoverageFiles(
         sm.sources = sm.sources.map(pathTransformer);
       }
 
-      coverage[p] = fileCoverage;
-    });
+      if (unifyFileMetadata === 'keep') {
+        coverage[p] = fileCoverage;
+      } else {
+        // When preparing for the final report, we sometimes end up in a situation
+        // in which the file metadata from different coverage reports are not exact matches
+        // for each other (different source maps, line/column numbers). That leads
+        // nyc to generate lower coverage numbers than it otherwise would.
+        // TODO: Investigate why this mismatch occurs (e.g.: some difference between
+        // running a package's own tests vs. using a package from another package's tests
+        // and then using the transpiled code from lib/ or dist/).
```

Comment:
If we are closing MONGOSH-1312 with this, maybe we open another for this todo?

Reply:
Sure, added a ticket + reference here! :+1: